server/src/main/java/org/elasticsearch/index/IndexSortConfig.java

```diff
@@ -173,7 +173,7 @@ public Sort buildIndexSort(Function<String, MappedFieldType> fieldTypeLookup,
             FieldSortSpec sortSpec = sortSpecs[i];
             final MappedFieldType ft = fieldTypeLookup.apply(sortSpec.field);
             if (ft == null) {
-                throw new IllegalArgumentException("unknown index sort field:[" + sortSpec.field + "]");
+                throw new IllegalArgumentException("unknown concrete index sort field: [" + sortSpec.field + "]");
```

Comment:
I thought it was worth changing the error here to make it explicit that you need a concrete field for index sorting.

Reply:
I agree the message may be unclear, but I am not sure that concrete would clarify it to users. Maybe we should rather add some suggestion after the error message "Please make sure that the field exists and is not a runtime field"